The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9911834048/job/27385354356) [(plain)](https://github.com/rust-lang/rust/commit/a792b450e3e5934578d826c2932ad5bc25ae4fc3/checks/27385354356/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling icu_provider_adapters v1.5.0
thread 'rustc' panicked at compiler/rustc_hir_analysis/src/collect.rs:1914:36:
$ident: found Synthetic
stack backtrace:
   0:     0x7f495dbe443f - <std[1e810906d2fadf8]::sys::backtrace::_print::DisplayBacktrace as core[839ef12c70fa2509]::fmt::Display>::fmt
   1:     0x7f495dc40c00 - core[839ef12c70fa2509]::fmt::write
   2:     0x7f495dbd8c39 - <std[1e810906d2fadf8]::sys::pal::unix::stdio::Stderr as std[1e810906d2fadf8]::io::Write>::write_fmt
   3:     0x7f495dbe420e - std[1e810906d2fadf8]::sys::backtrace::print
   4:     0x7f495dbe6eea - std[1e810906d2fadf8]::panicking::default_hook::{closure#1}
   5:     0x7f495dbe6b74 - std[1e810906d2fadf8]::panicking::default_hook
   6:     0x7f495f922367 - <alloc[8501d074461e147f]::boxed::Box<rustc_driver_impl[35eaa49fc270ff37]::install_ice_hook::{closure#0}> as core[839ef12c70fa2509]::ops::function::Fn<(&dyn for<'a, 'b> core[839ef12c70fa2509]::ops::function::Fn<(&'a std[1e810906d2fadf8]::panic::PanicHookInfo<'b>,), Output = ()> + core[839ef12c70fa2509]::marker::Send + core[839ef12c70fa2509]::marker::Sync, &std[1e810906d2fadf8]::panic::PanicHookInfo)>>::call
   7:     0x7f495dbe78d8 - std[1e810906d2fadf8]::panicking::rust_panic_with_hook
   8:     0x7f495dbe7497 - std[1e810906d2fadf8]::panicking::begin_panic_handler::{closure#0}
   9:     0x7f495dbe4a09 - std[1e810906d2fadf8]::sys::backtrace::__rust_end_short_backtrace::<std[1e810906d2fadf8]::panicking::begin_panic_handler::{closure#0}, !>
  11:     0x7f495dc3cbf3 - core[839ef12c70fa2509]::panicking::panic_fmt
  11:     0x7f495dc3cbf3 - core[839ef12c70fa2509]::panicking::panic_fmt
  12:     0x7f4963cf7243 - rustc_hir[33a1600977d30942]::hir::expect_failed::<&rustc_hir[33a1600977d30942]::hir::Node>
  13:     0x7f4963cfb340 - <rustc_hir[33a1600977d30942]::hir::Node>::expect_item
  14:     0x7f49606231fc - rustc_hir_analysis[38b88817ac6acbc1]::collect::rendered_precise_capturing_args
  15:     0x7f4961c3acd9 - rustc_query_impl[86461119850dbd66]::plumbing::__rust_begin_short_backtrace::<rustc_query_impl[86461119850dbd66]::query_impl::rendered_precise_capturing_args::dynamic_query::{closure#2}::{closure#0}, rustc_middle[7f972377108a344b]::query::erase::Erased<[u8; 16usize]>>
  16:     0x7f4961f3d2f1 - <rustc_query_impl[86461119850dbd66]::query_impl::rendered_precise_capturing_args::dynamic_query::{closure#2} as core[839ef12c70fa2509]::ops::function::FnOnce<(rustc_middle[7f972377108a344b]::ty::context::TyCtxt, rustc_span[25637ecbbdd08fed]::def_id::DefId)>>::call_once
  17:     0x7f4961c7e305 - rustc_query_system[aa6fb28dfa74b3be]::query::plumbing::try_execute_query::<rustc_query_impl[86461119850dbd66]::DynamicConfig<rustc_query_system[aa6fb28dfa74b3be]::query::caches::DefIdCache<rustc_middle[7f972377108a344b]::query::erase::Erased<[u8; 16usize]>>, false, false, false>, rustc_query_impl[86461119850dbd66]::plumbing::QueryCtxt, false>
  18:     0x7f4961f46e7a - rustc_query_impl[86461119850dbd66]::query_impl::rendered_precise_capturing_args::get_query_non_incr::__rust_end_short_backtrace
  19:     0x7f49631b302a - rustc_middle[7f972377108a344b]::query::plumbing::query_get_at::<rustc_query_system[aa6fb28dfa74b3be]::query::caches::DefIdCache<rustc_middle[7f972377108a344b]::query::erase::Erased<[u8; 16usize]>>>
  20:     0x7f49631c78f9 - <rustc_metadata[7e87f66a917431e]::rmeta::encoder::EncodeContext>::encode_precise_capturing_args
  21:     0x7f49631c4815 - <rustc_metadata[7e87f66a917431e]::rmeta::encoder::EncodeContext>::encode_def_ids
  22:     0x7f49631ba7b9 - <rustc_metadata[7e87f66a917431e]::rmeta::encoder::EncodeContext>::encode_crate_root
  23:     0x7f49631c965a - rustc_metadata[7e87f66a917431e]::rmeta::encoder::encode_metadata
  24:     0x7f4963111e30 - rustc_metadata[7e87f66a917431e]::fs::encode_and_write_metadata
  25:     0x7f495fcfe9c1 - rustc_interface[c1c3924bfe6fc8a2]::passes::start_codegen
  26:     0x7f495fbf9b12 - <rustc_interface[c1c3924bfe6fc8a2]::queries::Linker>::codegen_and_build_linker
  27:     0x7f495f8919be - <rustc_middle[7f972377108a344b]::ty::context::GlobalCtxt>::enter::<rustc_driver_impl[35eaa49fc270ff37]::run_compiler::{closure#0}::{closure#1}::{closure#6}, core[839ef12c70fa2509]::result::Result<core[839ef12c70fa2509]::option::Option<rustc_interface[c1c3924bfe6fc8a2]::queries::Linker>, rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>
  28:     0x7f495f9a6535 - <rustc_interface[c1c3924bfe6fc8a2]::queries::QueryResult<&rustc_middle[7f972377108a344b]::ty::context::GlobalCtxt>>::enter::<core[839ef12c70fa2509]::result::Result<core[839ef12c70fa2509]::option::Option<rustc_interface[c1c3924bfe6fc8a2]::queries::Linker>, rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>, rustc_driver_impl[35eaa49fc270ff37]::run_compiler::{closure#0}::{closure#1}::{closure#6}>
  29:     0x7f495f9259f1 - <rustc_interface[c1c3924bfe6fc8a2]::interface::Compiler>::enter::<rustc_driver_impl[35eaa49fc270ff37]::run_compiler::{closure#0}::{closure#1}, core[839ef12c70fa2509]::result::Result<core[839ef12c70fa2509]::option::Option<rustc_interface[c1c3924bfe6fc8a2]::queries::Linker>, rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>
  30:     0x7f495f90e1ed - rustc_span[25637ecbbdd08fed]::create_session_globals_then::<core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>, rustc_interface[c1c3924bfe6fc8a2]::util::run_in_thread_with_globals<rustc_interface[c1c3924bfe6fc8a2]::util::run_in_thread_pool_with_globals<rustc_interface[c1c3924bfe6fc8a2]::interface::run_compiler<core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>, rustc_driver_impl[35eaa49fc270ff37]::run_compiler::{closure#0}>::{closure#1}, core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>::{closure#0}, core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>::{closure#0}::{closure#0}::{closure#0}>
  31:     0x7f495f893702 - std[1e810906d2fadf8]::sys::backtrace::__rust_begin_short_backtrace::<rustc_interface[c1c3924bfe6fc8a2]::util::run_in_thread_with_globals<rustc_interface[c1c3924bfe6fc8a2]::util::run_in_thread_pool_with_globals<rustc_interface[c1c3924bfe6fc8a2]::interface::run_compiler<core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>, rustc_driver_impl[35eaa49fc270ff37]::run_compiler::{closure#0}>::{closure#1}, core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>::{closure#0}, core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>
  32:     0x7f495f8f16bf - <<std[1e810906d2fadf8]::thread::Builder>::spawn_unchecked_<rustc_interface[c1c3924bfe6fc8a2]::util::run_in_thread_with_globals<rustc_interface[c1c3924bfe6fc8a2]::util::run_in_thread_pool_with_globals<rustc_interface[c1c3924bfe6fc8a2]::interface::run_compiler<core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>, rustc_driver_impl[35eaa49fc270ff37]::run_compiler::{closure#0}>::{closure#1}, core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>::{closure#0}, core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>::{closure#0}::{closure#0}, core[839ef12c70fa2509]::result::Result<(), rustc_span[25637ecbbdd08fed]::ErrorGuaranteed>>::{closure#2} as core[839ef12c70fa2509]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  33:     0x7f495dbf3064 - <std[1e810906d2fadf8]::sys::pal::unix::thread::Thread>::new::thread_start
  35:     0x7f495da0a850 - <unknown>
  36:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.
error: the compiler unexpectedly panicked. this is a bug.

note: using internal features is not supported and expected to cause internal compiler errors when used incorrectly

warning: the ICE couldn't be written to `/checkout/rustc-ice-2024-07-12T17_46_50-16501.txt`: Read-only file system (os error 30)
note: rustc 1.81.0-nightly (deaf20af8 2024-07-12) running on x86_64-unknown-linux-gnu


note: compiler flags: --crate-type lib -C opt-level=3 -C embed-bitcode=no -C debug-assertions=on -C symbol-mangling-version=v0 -Z unstable-options -Z macro-backtrace -C split-debuginfo=off -Z unstable-options -C prefer-dynamic -C link-args=-Wl,-z,origin -C link-args=-Wl,-rpath,$ORIGIN/../lib -Z on-broken-pipe=kill -Z binary-dep-depinfo -Z tls-model=initial-exec -Z force-unstable-if-unmarked
note: some of the compiler flags provided by cargo are hidden

query stack during panic:
query stack during panic:
#0 [rendered_precise_capturing_args] computing precise capturing args for `graph::reference::<impl at compiler/rustc_data_structures/src/graph/reference.rs:17:1: 17:53>::{synthetic#0}`
error: could not compile `rustc_data_structures` (lib)
warning: build failed, waiting for other jobs to finish...
Build completed unsuccessfully in 0:05:41
+ set -e
+ set -e
+ cat /tmp/toolstate/toolstates.json
{"lld-wrapper":"test-fail"}
+ python3 ../x.py test --stage 2 check-tools
##[group]Building bootstrap
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
ERROR: Tool `book` was not recorded in tool state.
ERROR: Tool `nomicon` was not recorded in tool state.
ERROR: Tool `reference` was not recorded in tool state.
ERROR: Tool `rust-by-example` was not recorded in tool state.
ERROR: Tool `edition-guide` was not recorded in tool state.
ERROR: Tool `embedded-book` was not recorded in tool state.
  local time: Fri Jul 12 17:47:01 UTC 2024
  network time: Fri, 12 Jul 2024 17:47:01 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/compiler/rustc_hir_analysis/src/collect.rs
+++ b/compiler/rustc_hir_analysis/src/collect.rs
@@ -84,6 +84,7 @@ pub fn provide(providers: &mut Providers) {
         coroutine_kind,
         coroutine_for_closure,
         is_type_alias_impl_trait,
+        rendered_precise_capturing_args,
         ..*providers
     };
 }
@@ -1879,4 +1880,24 @@ fn is_type_alias_impl_trait<'tcx>(tcx: TyCtxt<'tcx>, def_id: LocalDefId) -> bool
         }
         _ => bug!("tried getting opaque_ty_origin for non-opaque: {:?}", def_id),
     }
+}
+
+fn rendered_precise_capturing_args<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    def_id: LocalDefId,
+) -> Option<&'tcx [Symbol]> {
+    if let Some(ty::ImplTraitInTraitData::Trait { opaque_def_id, .. }) =
+        tcx.opt_rpitit_info(def_id.to_def_id())
+    {
+        return tcx.rendered_precise_capturing_args(opaque_def_id);
+    }
+
+    tcx.hir_node_by_def_id(def_id).expect_item().expect_opaque_ty().bounds.iter().find_map(
+        |bound| match bound {
+            hir::GenericBound::Use(args, ..) => {
+                Some(&*tcx.arena.alloc_from_iter(args.iter().map(|arg| arg.name())))
+            }
+            _ => None,
+        },
+    )
 }

--- a/compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs
+++ b/compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs
@@ -72,6 +72,15 @@ impl<'a, 'tcx, T: Copy + Decodable<DecodeContext<'a, 'tcx>>> ProcessQueryValue<'
     }
 }
 
+impl<'a, 'tcx, T: Copy + Decodable<DecodeContext<'a, 'tcx>>>
+    ProcessQueryValue<'tcx, Option<&'tcx [T]>> for Option<DecodeIterator<'a, 'tcx, T>>
+{
+    #[inline(always)]
+    fn process_decoded(self, tcx: TyCtxt<'tcx>, _err: impl Fn() -> !) -> Option<&'tcx [T]> {
+        if let Some(iter) = self { Some(&*tcx.arena.alloc_from_iter(iter)) } else { None }
+    }
+}
+
 impl ProcessQueryValue<'_, Option<DeprecationEntry>> for Option<Deprecation> {
     #[inline(always)]
     fn process_decoded(self, _tcx: TyCtxt<'_>, _err: impl Fn() -> !) -> Option<DeprecationEntry> {
@@ -249,6 +258,7 @@ provide! { tcx, def_id, other, cdata,
             .process_decoded(tcx, || panic!("{def_id:?} does not have coerce_unsized_info"))) }
     mir_const_qualif => { table }
     rendered_const => { table }
+    rendered_precise_capturing_args => { table }
     asyncness => { table_direct }
     fn_arg_names => { table }
     coroutine_kind => { table_direct }

--- a/compiler/rustc_metadata/src/rmeta/encoder.rs
+++ b/compiler/rustc_metadata/src/rmeta/encoder.rs
@@ -1496,6 +1496,7 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
                 self.tables
                     .is_type_alias_impl_trait
                     .set(def_id.index, self.tcx.is_type_alias_impl_trait(def_id));
+                self.encode_precise_capturing_args(def_id);
             }
             if tcx.impl_method_has_trait_impl_trait_tys(def_id)
                 && let Ok(table) = self.tcx.collect_return_position_impl_trait_in_trait_tys(def_id)
@@ -1635,11 +1636,20 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
                     self.tables.assumed_wf_types_for_rpitit[def_id]
                         <- self.tcx.assumed_wf_types_for_rpitit(def_id)
                 );
+                self.encode_precise_capturing_args(def_id);
             }
         }
         if item.is_effects_desugaring {
             self.tables.is_effects_desugaring.set(def_id.index, true);
         }
+    }
+
+    fn encode_precise_capturing_args(&mut self, def_id: DefId) {
+        let Some(precise_capturing_args) = self.tcx.rendered_precise_capturing_args(def_id) else {
+            return;
+        };
+
+        record_array!(self.tables.rendered_precise_capturing_args[def_id] <- precise_capturing_args);
     }
 
     fn encode_mir(&mut self) {

--- a/compiler/rustc_metadata/src/rmeta/mod.rs
+++ b/compiler/rustc_metadata/src/rmeta/mod.rs
@@ -442,6 +442,7 @@ define_tables! {
     coerce_unsized_info: Table<DefIndex, LazyValue<ty::adjustment::CoerceUnsizedInfo>>,
     mir_const_qualif: Table<DefIndex, LazyValue<mir::ConstQualifs>>,
     rendered_const: Table<DefIndex, LazyValue<String>>,
+    rendered_precise_capturing_args: Table<DefIndex, LazyArray<Symbol>>,
     asyncness: Table<DefIndex, ty::Asyncness>,
     fn_arg_names: Table<DefIndex, LazyArray<Ident>>,
     coroutine_kind: Table<DefIndex, hir::CoroutineKind>,

--- a/compiler/rustc_middle/src/query/mod.rs
+++ b/compiler/rustc_middle/src/query/mod.rs
@@ -1261,6 +1261,7 @@ rustc_queries! {
         desc { |tcx| "looking up function parameter names for `{}`", tcx.def_path_str(def_id) }
         separate_provide_extern
     }
+
     /// Gets the rendered value of the specified constant or associated constant.
     /// Used by rustdoc.
     query rendered_const(def_id: DefId) -> &'tcx String {
@@ -1268,6 +1269,13 @@ rustc_queries! {
         desc { |tcx| "rendering constant initializer of `{}`", tcx.def_path_str(def_id) }
         separate_provide_extern
     }
+
+    /// Gets the rendered precise capturing args for an opaque for use in rustdoc.
+    query rendered_precise_capturing_args(def_id: DefId) -> Option<&'tcx [Symbol]> {
+        desc { |tcx| "rendering precise capturing args for `{}`", tcx.def_path_str(def_id) }
+        separate_provide_extern
+    }
+
     query impl_parent(def_id: DefId) -> Option<DefId> {
         desc { |tcx| "computing specialization parent impl of `{}`", tcx.def_path_str(def_id) }
         separate_provide_extern

--- a/src/librustdoc/clean/mod.rs
+++ b/src/librustdoc/clean/mod.rs
@@ -461,13 +461,7 @@ fn clean_projection<'tcx>(
     def_id: Option<DefId>,
 ) -> Type {
     if cx.tcx.is_impl_trait_in_trait(ty.skip_binder().def_id) {
-        let bounds = cx
-            .tcx
-            .explicit_item_bounds(ty.skip_binder().def_id)
-            .iter_instantiated_copied(cx.tcx, ty.skip_binder().args)
-            .map(|(pred, _)| pred)
-            .collect::<Vec<_>>();
-        return clean_middle_opaque_bounds(cx, bounds);
+        return clean_middle_opaque_bounds(cx, ty.skip_binder().def_id, ty.skip_binder().args);
     }
 
     let trait_ = clean_trait_ref_with_constraints(
@@ -2243,13 +2237,7 @@ pub(crate) fn clean_middle_ty<'tcx>(
                 *cx.current_type_aliases.entry(def_id).or_insert(0) += 1;
                 // Grab the "TraitA + TraitB" from `impl TraitA + TraitB`,
                 // by looking up the bounds associated with the def_id.
-                let bounds = cx
-                    .tcx
-                    .explicit_item_bounds(def_id)
-                    .iter_instantiated_copied(cx.tcx, args)
-                    .map(|(bound, _)| bound)
-                    .collect::<Vec<_>>();
-                let ty = clean_middle_opaque_bounds(cx, bounds);
+                let ty = clean_middle_opaque_bounds(cx, def_id, args);
                 if let Some(count) = cx.current_type_aliases.get_mut(&def_id) {
                     *count -= 1;
                     if *count == 0 {
@@ -2272,12 +2260,20 @@ pub(crate) fn clean_middle_ty<'tcx>(
 
 fn clean_middle_opaque_bounds<'tcx>(
     cx: &mut DocContext<'tcx>,
-    bounds: Vec<ty::Clause<'tcx>>,
+    impl_trait_def_id: DefId,
+    args: ty::GenericArgsRef<'tcx>,
 ) -> Type {
     let mut has_sized = false;
+
+    let bounds: Vec<_> = cx
+        .tcx
+        .explicit_item_bounds(impl_trait_def_id)
+        .iter_instantiated_copied(cx.tcx, args)
+        .collect();
+
     let mut bounds = bounds
         .iter()
-        .filter_map(|bound| {
+        .filter_map(|(bound, _)| {
             let bound_predicate = bound.kind();
             let trait_ref = match bound_predicate.skip_binder() {
                 ty::ClauseKind::Trait(tr) => bound_predicate.rebind(tr.trait_ref),
@@ -2296,7 +2292,7 @@ fn clean_middle_opaque_bounds<'tcx>(
 
             let bindings: ThinVec<_> = bounds
                 .iter()
-                .filter_map(|bound| {
+                .filter_map(|(bound, _)| {
                     if let ty::ClauseKind::Projection(proj) = bound.kind().skip_binder() {
                         if proj.projection_term.trait_ref(cx.tcx) == trait_ref.skip_binder() {
                             Some(AssocItemConstraint {
@@ -2334,6 +2330,10 @@ fn clean_middle_opaque_bounds<'tcx>(
     // Since all potential trait bounds are at the front we can just check the first bound.
     if bounds.first().map_or(true, |b| !b.is_trait_bound()) {
         bounds.insert(0, GenericBound::sized(cx));
+    }
+
+    if let Some(args) = cx.tcx.rendered_precise_capturing_args(impl_trait_def_id) {
+        bounds.push(GenericBound::Use(args.to_vec()));
     }
 
     ImplTrait(bounds)

--- a/tests/rustdoc/auxiliary/precise-capturing.rs
+++ b/tests/rustdoc/auxiliary/precise-capturing.rs
@@ -1,0 +1,7 @@
+#![feature(precise_capturing)]
+
+pub fn cross_crate_empty() -> impl Sized + use<> {}
+
+pub fn cross_crate_missing() -> impl Sized {}
+
+pub fn cross_crate_args<'a, T, const N: usize>() -> impl Sized + use<'a, T, N> {}

--- a/tests/rustdoc/impl-trait-precise-capturing.rs
+++ b/tests/rustdoc/impl-trait-precise-capturing.rs
@@ -1,5 +1,9 @@
+//@ aux-build:precise-capturing.rs
+
 #![crate_name = "foo"]
 #![feature(precise_capturing)]
+
+extern crate precise_capturing;
 
 //@ has foo/fn.two.html '//section[@id="main-content"]//pre' "-> impl Sized + use<'b, 'a>"
 pub fn two<'a, 'b, 'c>() -> impl Sized + use<'b, 'a /* no 'c */> {}
@@ -12,3 +16,12 @@ pub fn none() -> impl Sized + use<> {}
 
 //@ has foo/fn.first.html '//section[@id="main-content"]//pre' "-> impl use<> + Sized"
 pub fn first() -> impl use<> + Sized {}
+
+//@ has foo/fn.cross_crate_empty.html '//section[@id="main-content"]//pre' "-> impl Sized + use<>"
+pub use precise_capturing::cross_crate_empty;
+
+//@ matches foo/fn.cross_crate_missing.html '//section[@id="main-content"]//pre' "-> impl Sized$"
+pub use precise_capturing::cross_crate_missing;
+
+//@ has foo/fn.cross_crate_args.html '//section[@id="main-content"]//pre' "-> impl Sized + use<'a, T, N>"
+pub use precise_capturing::cross_crate_args;

